src/app/Dashboard/dashboard-utils.tsx

```diff
@@ -0,0 +1,230 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import cryostatLogo from '@app/assets/cryostat_icon_rgb_default.svg';
+import cryostatLogoDark from '@app/assets/cryostat_icon_rgb_reverse.svg';
+import { FeatureLevel } from '@app/Shared/Services/Settings.service';
+import { withThemedIcon } from '@app/utils/withThemedIcon';
+import { LabelProps, gridSpans } from '@patternfly/react-core';
+import { FileIcon, UnknownIcon, UserIcon } from '@patternfly/react-icons';
+import { nanoid } from '@reduxjs/toolkit';
+import React from 'react';
+import { Observable } from 'rxjs';
+
+export const DEFAULT_DASHBOARD_NAME = 'Default';
+export const DRAGGABLE_REF_KLAZZ = `draggable-ref`;
+export const LAYOUT_TEMPLATE_DESCRIPTION_WORD_LIMIT = 100;
+
+export const DashboardLayoutNamePattern = /^[a-zA-Z0-9_.-]+( [a-zA-Z0-9_.-]+)*$/;
+export const LayoutTemplateDescriptionPattern = /^[a-zA-Z0-9\s.,\-'";?!@#$%^&*()[\]_+=:{}]*$/;
+
+export interface CardConfig {
+  id: string;
+  name: string;
+  span: gridSpans;
+  props: object;
+}
+
+export type SerialCardConfig = Omit<CardConfig, 'id'>;
+
+export const mockSerialCardConfig: SerialCardConfig = {
+  name: 'Default',
+  span: 12,
+  props: {},
+};
+export interface DashboardLayout {
+  name: string;
+  cards: CardConfig[];
+  favorite: boolean;
+}
+
+export type SerialDashboardLayout = Omit<DashboardLayout, 'name' | 'cards'> & { cards: SerialCardConfig[] };
+
+// only name and vendor are needed to identify a template
+export type LayoutTemplateRecord = Pick<LayoutTemplate, 'name' | 'vendor'>;
+
+export enum LayoutTemplateVersion {
+  'v2.3' = 'v2.3',
+}
+
+export enum LayoutTemplateVendor {
+  BLANK = 'Blank',
+  CRYOSTAT = 'Cryostat',
+  USER = 'User-submitted',
+}
+
+export interface LayoutTemplate {
+  name: string;
+  description: string;
+  cards: SerialCardConfig[];
+  version: LayoutTemplateVersion;
+  vendor: LayoutTemplateVendor;
+}
+
+export type SerialLayoutTemplate = Omit<LayoutTemplate, 'vendor'>;
+
+export const mockSerialLayoutTemplate: SerialLayoutTemplate = {
+  name: 'Default',
+  description: 'Default.',
+  cards: [] as SerialCardConfig[],
+  version: LayoutTemplateVersion['v2.3'],
+};
+
+export interface LayoutTemplateController {
+  selectedTemplate: LayoutTemplate | undefined;
+  setSelectedTemplate: (template: React.SetStateAction<LayoutTemplate | undefined>) => void;
+  isUploadModalOpen: boolean;
+  setIsUploadModalOpen: (isOpen: React.SetStateAction<boolean>) => void;
+}
+
+// use a provider
+export const LayoutTemplateContext = React.createContext<LayoutTemplateController>({
+  selectedTemplate: undefined,
+  setSelectedTemplate: () => undefined,
+  isUploadModalOpen: false,
+  setIsUploadModalOpen: () => undefined,
+});
+
+export const iconify = (vendor: LayoutTemplateVendor): React.ReactNode => {
```

Comment:
The user icon is not shown....?

![Screenshot from 2023-04-12 21-06-14](https://user-images.githubusercontent.com/68053619/231619404-ab3ff097-e60a-4794-8283-03354b3fc6d3.png)

Reply:
Ahh never mind, I am using the old templates.

Reply:
Worked perfectly haha.